Minwoolee/callingserverv3 fixclients

minwoolee-msft opened the pull request:
From Feedbacks and suggestion, Here are the latest updates for Server Calling: 

- Have multiple Clients for each groups of methods: CallConnectionClient, CallRecordingClient, CallContentclient
- New Clients can be fetched from Get[ClientName], from root service client
- Updated other download helpers to work with CallRecordingClient
- CallConnectionProperties is now separate object that only contains data of the call. All methods are moved into CallConnectionClient
- AnswerCall, MakeCall, GetCall now returns CallConnectionProperties instead of CallConnectionClient
- Removed reason as input from TerminateCall (this was never used)
- GetParticipants now returns participants in IReadOnlyList
- Various Request objects are now internal, hidden from contoso (i.e. RejectCallRequest)
- Dropped unnecessary suffix, like Dto and Model (i.e.  CallConnectionStateModel ->  CallConnectionState)
- Options (such as AddParticipantsOptions) constructor now only takes optional inputs. All other options updated to be similar
- CallLocator now updated to be Abstract class. Derived classes (GroupCallLocator & ServerCallLocator) Added.
- Moved most of Models into correct namespace
- Deleted unused models
- Various small typeo fixes

minwoolee-msft committed:
Initial Commit from feedbacks:

- Have multiple Clients for each grops of methods: CallConnectionClient, CallRecordingClient, CallContentclient
- New Clients can be fetched from Get<ClientName>, from root service client
- Updated other download helpers to work with CallRecordingClient
- CallConnectionProperties is now seperate object that only contains data of the call. All methods are moved into CallConnectionClient
- AnswerCall, MakeCall, GetCall now returns CallConnectionProperties instead of CallConnectionClient
- Removed reason as input from TerminateCall (this was never used)
- GetParticipants now returns participants in IReadOnlyList
- Various Request objects are now internal, hidden from contoso (i.e. RejectCallRequest)
- Dropped unnessary suffixs, like Dto and Model (i.e.  CallConnectionStateModel ->  CallConnectionState)
- Options (such as AddParticipantsOptions) constructor now only takes optional inputs. All other options updated to be similar
- CallLocator now updated to be Abstract class. Derived classes (GroupCallLocator & ServerCallLocator) Added.

minwoolee-msft committed:
Updated APi file

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Communication.CallingServer](https://apiview.dev/Assemblies/Review/d034a0e806f94bccb550908bdeae409e)

minwoolee-msft committed:
Fixing the tests

minwoolee-msft committed:
Updated loggings

minwoolee-msft committed:
AddParticipant -> AddParticipants

minwoolee-msft committed:
Updated API

yyw-msft commented on sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:
```diff
@@ -425,245 +408,15 @@ public virtual Response<CallConnection> GetCall(string callConnectionId, Cancell
-        /// <param name="cancellationToken">The cancellation token.</param>
-        public virtual Response ResumeRecording(string recordingId, CancellationToken cancellationToken = default)
+        /// <summary> Initializes a new instance of CallConnectionClient. <see cref="GetCallConnectionClient"/>.</summary>
+        /// <param name="callConnectionId"> The thread id for the ChatThreadClient instance. </param>
```
update callConnectionId description?

yyw-msft commented on sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:
```diff
@@ -672,237 +425,14 @@ public virtual Response ResumeRecording(string recordingId, CancellationToken ca
-        /// a failure occurs.
-        /// </remarks>
-        public virtual Response DeleteRecording(Uri deleteEndpoint, CancellationToken cancellationToken = default)
+        /// <summary> Initializes a new instance of CallConnectionClient. <see cref="CallRecordingClient"/>.</summary>
```
Initialize new CallRecordingClient?

minwoolee-msft committed:
- Fixing various typeos
- Updated summaries
- Moved Content portions into proper client
- Renamed TransferOptions to TransferCalltoParticipantOptions, to be more specific
- Moved around models into proper namespace

minwoolee-msft committed:
PlayOptions now internal

minwoolee-msft committed:
API Update

minwoolee-msft committed:
Updated RemoveParticipants to accept RemoveParticipantsOptions
